File: week_1/03_02_find_alphabet_occurrence.py
def find_max_occurred_alphabet(string):
    alphabet_occurence_array=[0]*26 #알파벳 빈도수를 담는 배열

    for char in string:
        if not char.isalpha(): #만약 알파벳이 아니라면
            continue
        arr_idx=ord(char) - ord('a') #아스키 코드를 이용하여 인덱스를 구함
        alphabet_occurence_array[arr_idx] += 1 #그 인덱스에 맞는 알파벳 카운트

        #이제 알파벳별 빈도수 체크
        max_occurrence = 0
        max_alphabet_index = 0
        for i in range(len(alphabet_occurence_array)):
            alphabet_occurrence=alphabet_occurence_array[i] #알파벳 빈도
            if alphabet_occurrence > max_occurrence:
                max_occurrence = alphabet_occurrence
                max_alphabet_index=i #최빈 알파벳 인덱스 교체
    return chr(max_alphabet_index+ord('a'))
    
result = find_max_occurred_alphabet
print("정답 = a 현재 풀이 값 =", result("Hello my name is sparta"))
print("정답 = a 현재 풀이 값 =", result("Sparta coding club"))
print("정답 = s 현재 풀이 값 =", result("best of best sparta"))

File: week_1/03_02_find_alphabet_occurrence.cpp
#include <iostream>
#include <string>
using namespace std;

char find_max_occurred_alphabet(string str) { //기존 알파벳별 빈도수 체크 배열이용 함수
	int alphabet_occurrence_array[26] = { 0, }; //알파벳별 빈도수를 저장하는 배열
	for (int i = 0; i < str.length(); i++) {
		if (!isalpha(str[i])) //해당 인덱스의 문자열에 알파벳이 아닌 경우
			continue;
		int arr_idx = (str[i] - 'a'); //아스키 코드 이용
		alphabet_occurrence_array[arr_idx] += 1;
	}
	int max_occurrence = 0; //최빈 변수
	int max_alphabet_idx = 0;
	
	for (int j = 0; j < 26; j++) { //알파벳 빈도수 최대 인덱스 찾기 위에서 사용한 배열 이용
		int alphabet_occurrence = alphabet_occurrence_array[j];
		if (max_occurrence < alphabet_occurrence) { // 크다면
			//교체 
			max_occurrence = alphabet_occurrence;
			max_alphabet_idx = j;
		}
	}
	return (max_alphabet_idx +'a'); //인덱스에 +'a'를 더하면 해당 알파벳으로 변환
}

int main(void) {
	cout << find_max_occurred_alphabet("Hello my name is sparta") << '\n';
	cout << find_max_occurred_alphabet("Sparta coding club") << '\n';
	cout << find_max_occurred_alphabet("best of best sparta") << '\n';
	return 0;
}
